src/test/java/org/jabref/logic/citationstyle/CitationStyleGeneratorTest.java

```diff
@@ -166,85 +170,427 @@ void testHandleCrossRefFields() {
-
-        BibDatabaseContext bibDatabaseContext = new BibDatabaseContext(new BibDatabase(List.of(entry)));
-        bibDatabaseContext.setMode(BibDatabaseMode.BIBTEX);
-
-        String citation = CitationStyleGenerator.generateCitation(entry, CitationStyle.getDefault().getSource(), CitationStyleOutputFormat.TEXT, bibDatabaseContext, bibEntryTypesManager);
-        assertEquals(expected, citation);
-    }
-
-    @Test
-    void testBiblatexWithIssue() {
-        BibEntry entry = new BibEntry(StandardEntryType.Article);
-        entry.setField(StandardField.AUTHOR, "Last, First and\nDoe, Jane");
-        entry.setField(StandardField.ISSUE, "7");
-
+    static Stream<Arguments> testCslMapping() {
         // if the default citation style changes this has to be modified
-        String expected = "[1]F. Last and J. Doe, no. 7.\n";
-
-        BibDatabaseContext bibDatabaseContext = new BibDatabaseContext(new BibDatabase(List.of(entry)));
-        bibDatabaseContext.setMode(BibDatabaseMode.BIBLATEX);
-
-        String citation = CitationStyleGenerator.generateCitation(entry, CitationStyle.getDefault().getSource(), CitationStyleOutputFormat.TEXT, bibDatabaseContext, bibEntryTypesManager);
-        assertEquals(expected, citation);
+        return Stream.of(
+                Arguments.of(
+                        "[1]F. Last and J. Doe, no. 28.\n",
+                        BibDatabaseMode.BIBTEX,
+                        new BibEntry(StandardEntryType.Article)
+                                .withField(StandardField.AUTHOR, "Last, First and\nDoe, Jane")
+                                .withField(StandardField.NUMBER, "28"),
+                        "ieee.csl"),
+                Arguments.of(
+                        "[1]F. Last and J. Doe, no. 7.\n",
+                        BibDatabaseMode.BIBLATEX,
+                        new BibEntry(StandardEntryType.Article)
+                                .withField(StandardField.AUTHOR, "Last, First and\nDoe, Jane")
+                                .withField(StandardField.ISSUE, "7"),
+                        "ieee.csl"),
+                Arguments.of(
+                        "[1]F. Last and J. Doe, no. 28.\n",
+                        BibDatabaseMode.BIBLATEX,
+                        new BibEntry(StandardEntryType.Article)
+                                .withField(StandardField.AUTHOR, "Last, First and Doe, Jane")
+                                .withField(StandardField.NUMBER, "28"),
+                        "ieee.csl"),
+                Arguments.of(
+                        "[1]F. Last and J. Doe, no. 7, Art. no. 28.\n",
+                        BibDatabaseMode.BIBLATEX,
+                        new BibEntry(StandardEntryType.Article)
+                                .withField(StandardField.AUTHOR, "Last, First and\nDoe, Jane")
+                                .withField(StandardField.ISSUE, "7")
+                                .withField(StandardField.NUMBER, "28"),
```

Comment:
Replace `.withField(StandardField.NUMBER, "28"),` with `.withField(StandardField.EID, "55"),`

Reason: "Art. no. xx" is only necessary for the EID field. See page "II. Style—13" at https://journals.ieeeauthorcenter.ieee.org/wp-content/uploads/sites/7/IEEE_Reference_Guide.pdf